### ISSUE TITLE
We need an autosave feature so if Lizzie completely crashes, the game resumes from last input

### ISSUE THREAD

**Issue by hydrogenpi** [featurecat/lizzie#140]
Some people are already using this to play against strong pros and its frustrating when it crashes mid game... 

We need an autosave feature so if Lizzie completely crashes, the game resumes from last input

Right now the only workaround is to save a sgf every few moves. Not good. 

Many times the Leelaz.exe within Lizzie will exception out and crash mid game. 

Lets do an autosave like how Word and etc does it please. Auto recovery too
it should save something to temp file/folder per move that is made as it is made so that a full recovery can be made in the event of an unexpected crash or shutdown

**Comment by featurecat** [comment]
what? Leela zero has never crashed for me. did you compile it yourself? I also use lizzie a lot so I've tested this quite a bit.

**Comment by hydrogenpi** [comment]
![image](https://user-images.githubusercontent.com/35181396/39282707-72bd8148-48bf-11e8-9b50-a5259c4ad24e.png)

**Comment by hydrogenpi** [comment]
![image](https://user-images.githubusercontent.com/35181396/39282733-8e000c46-48bf-11e8-8c11-4e655d9ebf0f.png)

**Comment by hydrogenpi** [comment]
![image](https://user-images.githubusercontent.com/35181396/39282788-bdcb30d6-48bf-11e8-8761-6b35aad68981.png)

**Comment by Ka-zam** [comment]
I maintain a Win 64-bit build with Lizzie support that tracks next branch here:
https://github.com/Ka-zam/leela-zero/releases

It has never crashed for me since SMP and uct pointer fixed were pushed.

**Comment by aerisnju** [comment]
I've never experienced such a crash, but one of my friends has. After my checking, his computer had only 8GB memories but two GPU cards installed. The memory for leelaz would be exhausted soon after analyzing for several minutes, especially when using a larger network file(e.g. 20b256f).
I suggested adding some memories and the crash frequency has been reduced significantly. So the amount of memory maybe one of the causes.

**Comment by AncalagonX** [comment]
We definitely need this feature. Non-Lizzie related computer crashes (GPU overheat, driver crash, etc.) can cause the entire game to be lost. Things like this have happened several times to me, and I would greatly appreciate an auto-save feature no matter how simple its implementation.

**Comment by baduk1** [comment]
Yes, would be good. My lizzie crushes sometimes when I run ELF weights.

**Comment by hydrogenpi** [comment]
Memory isn't the issue, I have more than 200GB of RAM and not even 10% was used when it crashes

**Comment by OlivierBlanvillain** [comment]
I think we can considered this to be solved after #298. If there are still thinks to improve on top of that please make a new issue!
